src/main.rs

```diff
@@ -41,18 +42,23 @@ fn main() {
 }
 
 /// Try to read `shell.nix` from the current working dir.
-fn get_shell_nix() -> Result<NixFile, ExitError> {
+fn get_shell_nix(shellfile: PathBuf) -> Result<NixFile, ExitError> {
     let current_dir_msg = || match env::current_dir() {
         Err(_) => String::from(""),
         Ok(pb) => format!(" ({})", pb.display()),
     };
     // use shell.nix from cwd
-    Ok(NixFile::from(locate_file::in_cwd("shell.nix").map_err(
+    let shellfile_str = match shellfile.to_str() {
+        Some(i) => i,
+        _ => "",
+    };
+    Ok(NixFile::from(locate_file::in_cwd(shellfile_str).map_err(
         |_| {
             ExitError::errmsg(format!(
-                "There is no `shell.nix` in the current directory{}\n\
+                "There is no `{}` in the current directory{}\n\
```

Comment:
It also works with an absolute path like `/home/user/shell.nix`.

Then the error message would say `"There is no `/home/user/shell.nix` in the current directory"`.

Instead, when the user provides a `shellfile`, it should always print a different message, like `"{} does not exist.`.